src/controller/java/BUILD.gn

```diff
@@ -122,12 +114,28 @@ android_library("java") {
     "zap-generated/chip/devicecontroller/ClusterWriteMapping.java",
   ]
 
+  if (build_java_matter_controller) {
```

Comment:
It is better to use flag name 'build_java_api' instead of 'build_java_matter_controller' since we are build the CHIPController.jar for any Java example app, not just java-matter-controller

Reply:
we would expect only have one java example app.